molgenis-data-mapper/src/main/java/org/molgenis/data/mapper/algorithmgenerator/generator/AbstractCategoryAlgorithmGenerator.java

```diff
@@ -12,18 +16,20 @@
 
 import com.google.common.base.Preconditions;
 
-public abstract class CategoryAlgorithmGenerator
+public abstract class AbstractCategoryAlgorithmGenerator implements AlgorithmGenerator
 {
 	private final DataService dataService;
 
-	public CategoryAlgorithmGenerator(DataService dataService)
+	public AbstractCategoryAlgorithmGenerator(DataService dataService)
 	{
 		this.dataService = Preconditions.checkNotNull(dataService);
 	}
 
-	public abstract boolean isSuitable(AttributeMetaData targetAttribute, List<AttributeMetaData> sourceAttributes);
-
-	public abstract String generate(AttributeMetaData targetAttribute, List<AttributeMetaData> sourceAttributes);
+	boolean isReferenceDataType(AttributeMetaData attribute)
```

Comment:
In this case, we only want to generate algorithms for xref and categorical reference types